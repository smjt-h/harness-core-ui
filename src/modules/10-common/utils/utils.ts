/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}

export const isCommunityPlan = (): boolean => false

export const isOnPrem = (): boolean => window.deploymentType === 'ON_PREM'

export const focusOnNode = (node: HTMLElement): void => {
  const oldTabIndex = node.tabIndex
  node.tabIndex = -1
  node.focus()
  node.tabIndex = oldTabIndex
}

const HOTJAR_SUPPRESSION_ATTR = 'data-hj-suppress'

// Utility to add `data-hj-suppress` into a collection of elements to
// suppress data from HotJar recording
// @see https://bit.ly/3rCgpOY
export const suppressHotJarRecording = (elements: Element[] | null | undefined): void => {
  if (window.hj) {
    elements?.forEach?.((e: Element) => {
      if (!e.hasAttribute(HOTJAR_SUPPRESSION_ATTR)) {
        e.setAttribute(HOTJAR_SUPPRESSION_ATTR, 'true')
      }
    })
  }
}

// Utility to generate { 'data-hj-suppress': true } attribute if HotJar is available
export const addHotJarSuppressionAttribute = (): { [HOTJAR_SUPPRESSION_ATTR]: boolean } | undefined =>
  window.hj ? { [HOTJAR_SUPPRESSION_ATTR]: true } : undefined
