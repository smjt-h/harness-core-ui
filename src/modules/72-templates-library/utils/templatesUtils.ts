/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { isEmpty, isEqual, startsWith } from 'lodash-es'
import { RUNTIME_INPUT_VALUE } from '@wings-software/uicore'
import get from 'lodash-es/get'
import type { UseStringsReturn } from 'framework/strings'
import { Scope } from '@common/interfaces/SecretsInterface'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import type { NGTemplateInfoConfig, TemplateSummaryResponse } from 'services/template-ng'

export enum TemplateType {
  Step = 'Step',
  Stage = 'Stage',
  Pipeline = 'Pipeline',
  Service = 'Service',
  Infrastructure = 'Infrastructure',
  StepGroup = 'StepGroup',
  Execution = 'Execution',
  MonitoredService = 'MonitoredService'
}

export const AllTemplatesTypes = 'All'

export interface TemplateTypeOption {
  label: string
  value: string
  disabled?: boolean
}

export const getScopeBasedQueryParams = (
  { accountId, projectIdentifier, orgIdentifier }: ProjectPathProps,
  scope: Scope
) => {
  return {
    accountIdentifier: accountId,
    projectIdentifier: scope === Scope.PROJECT ? projectIdentifier : undefined,
    orgIdentifier: scope === Scope.PROJECT || scope === Scope.ORG ? orgIdentifier : undefined
  }
}

export const getAllowedTemplateTypes = (
  getString: UseStringsReturn['getString'],
  module?: string,
  isPipelineTemplateEnabled?: boolean
): TemplateTypeOption[] => {
  const AllowedTemplateTypes = [
    {
      label: getString('step'),
      value: TemplateType.Step,
      disabled: false
    },
    {
      label: getString('common.stage'),
      value: TemplateType.Stage,
      disabled: false
    },
    {
      label: getString('common.pipeline'),
      value: TemplateType.Pipeline,
      disabled: !isPipelineTemplateEnabled
    },
    {
      label: getString('service'),
      value: TemplateType.Service,
      disabled: true
    },
    {
      label: getString('infrastructureText'),
      value: TemplateType.Infrastructure,
      disabled: true
    },
    {
      label: getString('stepGroup'),
      value: TemplateType.StepGroup,
      disabled: true
    },
    {
      label: getString('executionText'),
      value: TemplateType.Execution,
      disabled: true
    }
  ]
  if (module === 'cv') {
    AllowedTemplateTypes.push({
      label: getString('connectors.cdng.monitoredService.label'),
      value: TemplateType.MonitoredService,
      disabled: false
    })
    return AllowedTemplateTypes.sort((a, b) => Number(a.disabled) - Number(b.disabled))
  }
  return AllowedTemplateTypes
}

export const getVersionLabelText = (template: TemplateSummaryResponse, getString: UseStringsReturn['getString']) => {
  return isEmpty(template.versionLabel)
    ? getString('templatesLibrary.alwaysUseStableVersion')
    : template.stableTemplate
    ? getString('templatesLibrary.stableVersion', { entity: template.versionLabel })
    : template.versionLabel
}

export const getTemplateInputsCount = (temp: NGTemplateInfoConfig): number =>
  (JSON.stringify(temp)?.match(/<\+input>/g) || []).length

export const hasSameRunTimeInputs = (a: any, b: any): boolean => {
  if (isEqual(a, b)) {
    return true
  } else if (startsWith(a, RUNTIME_INPUT_VALUE) || startsWith(b, RUNTIME_INPUT_VALUE)) {
    return false
  } else if (typeof a !== 'object' || typeof b !== 'object') {
    return getTemplateInputsCount(a) === 0 && getTemplateInputsCount(b) === 0
  } else {
    if (Array.isArray(a) && Array.isArray(b)) {
      return a
        .filter(item => {
          for (const val in Object.values(item)) {
            if (startsWith(val, RUNTIME_INPUT_VALUE)) {
              return true
            }
          }
          return false
        })
        .every(item1 => {
          for (const item2 in b) {
            if (Object.entries(item1).every(([key, value]) => get(item2, key) === value)) {
              return true
            }
          }
          return false
        })
    } else if (!Array.isArray(a) && !Array.isArray(b)) {
      return Object.keys(a).every(k => hasSameRunTimeInputs(a[k], b[k]))
    } else {
      return getTemplateInputsCount(a) === 0 && getTemplateInputsCount(b) === 0
    }
  }
}
