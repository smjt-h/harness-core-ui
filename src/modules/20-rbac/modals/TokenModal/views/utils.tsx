/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getReadableDateTime } from '@common/utils/dateUtils'

export const DATE_FORMAT = 'MM/DD/YYYY'

export const hasNoExpiryDate = (value?: number): boolean => {
  return getReadableDateTime(value, DATE_FORMAT) === '12/31/2099'
}

export const getSelectedExpiration = (value?: number): string => {
  if (!value) {
    return '30'
  } else if (hasNoExpiryDate(value)) {
    return '-1'
  }
  return 'custom'
}

export const getSelectedExpirationDate = (value: string): string => {
  switch (value) {
    case '30':
    case '90':
    case '180':
      return getReadableDateTime(new Date().setDate(new Date().getDate() + parseInt(value)), DATE_FORMAT)
    case '-1':
      return '12/31/2099'
    default:
      return ''
  }
}

export const getExpirationDate = (value?: number): string => {
  if (value) {
    return getReadableDateTime(value, DATE_FORMAT)
  } else {
    return getSelectedExpirationDate(getSelectedExpiration(value))
  }
}
