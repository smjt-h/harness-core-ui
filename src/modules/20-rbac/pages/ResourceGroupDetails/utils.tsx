/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import produce from 'immer'
import type { SelectOption } from '@wings-software/uicore'
import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import { isDynamicResourceSelector, SelectionType } from '@rbac/utils/utils'
import type {
  ResourceSelector,
  ResponseResourceTypeDTO,
  ScopeSelector,
  ResourceGroupV2,
  ResourceSelectorV2,
  ResourceGroupV2Request,
  ResourceFilter
} from 'services/resourcegroups'
import { Scope } from '@common/interfaces/SecretsInterface'
import type { UseStringsReturn } from 'framework/strings'
import RbacFactory from '@rbac/factories/RbacFactory'
import { getScopeFromDTO } from '@common/components/EntityReference/EntityReference'

export enum SelectorScope {
  CURRENT = 'CURRENT',
  INCLUDE_CHILD_SCOPES = 'INCLUDE_CHILD_SCOPES',
  CUSTOM = 'CUSTOM'
}

enum ValidatorTypes {
  BY_RESOURCE_IDENTIFIER = 'BY_RESOURCE_IDENTIFIER',
  BY_RESOURCE_TYPE = 'BY_RESOURCE_TYPE',
  BY_RESOURCE_TYPE_INCLUDING_CHILD_SCOPES = 'BY_RESOURCE_TYPE_INCLUDING_CHILD_SCOPES'
}

export const getSelectedResourcesMap = (
  resourceTypes: ResourceType[],
  resourceFilter?: ResourceFilter
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()
  if (resourceFilter?.includeAllResources) {
    resourceTypes.forEach(resourceType => {
      map.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    })
  } else {
    resourceFilter?.resources?.map(resourceSelector => {
      if (resourceSelector.identifiers?.length) {
        map.set(resourceSelector.resourceType as ResourceType, resourceSelector.identifiers)
      } else {
        map.set(resourceSelector.resourceType as ResourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
      }
    })
  }

  return map
}

export const validateResourceSelectors = (value: ResourceSelector[]): ResourceType[] => {
  return value
    .filter(val => {
      if (!isDynamicResourceSelector(val.type)) {
        if (val.identifiers?.length < 1) {
          return true
        }
      }
      return false
    })
    .map(val => val.resourceType)
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelectorV2[] => {
  const resourceSelectors: ResourceSelectorV2[] = []
  map.forEach((value, key) => {
    if (isDynamicResourceSelector(value)) {
      resourceSelectors.push({
        resourceType: key
      })
    } else {
      resourceSelectors.push({
        resourceType: key,
        identifiers: value as string[]
      })
    }
  })
  return resourceSelectors
}

export const computeResourceMapOnChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  resourceType: ResourceType,
  isAdd: boolean,
  identifiers?: string[]
): void => {
  if (identifiers) {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, identifiers)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          if (Array.isArray(draft.get(resourceType))) {
            draft.set(
              resourceType,
              (draft.get(resourceType) as string[]).filter(el => !identifiers.includes(el))
            )
          }
          if (draft.get(resourceType)?.length === 0) {
            draft.delete(resourceType)
          }
        })
      )
    }
  } else {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.delete(resourceType)
        })
      )
    }
  }
}

export const computeResourceMapOnMultiChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  types: ResourceType[],
  isAdd: boolean
): void => {
  if (isAdd) {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      })
    )
  } else {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.delete(resourceType)
        })
      })
    )
  }
}

export const getScopeLabelFromApi = (
  getString: UseStringsReturn['getString'],
  scope: Scope,
  resourceGroup: ResourceGroupV2
): string => {
  const selectorScope = getScopeType(resourceGroup)
  const dropDownItems = getScopeDropDownItems(scope, getString)
  const option = dropDownItems.filter(item => item.value === selectorScope)
  return option.length ? option[0].label : ''
}

export const getScopeDropDownItems = (scope: Scope, getString: UseStringsReturn['getString']): SelectOption[] => {
  switch (scope) {
    case Scope.PROJECT:
      return [{ label: getString('rbac.scopeItems.projectOnly'), value: SelectorScope.CURRENT }]
    case Scope.ORG:
      return [
        { label: getString('rbac.scopeItems.orgOnly'), value: SelectorScope.CURRENT },
        { label: getString('rbac.scopeItems.orgAll'), value: SelectorScope.INCLUDE_CHILD_SCOPES }
      ]
    case Scope.ACCOUNT:
    default:
      return [
        { label: getString('rbac.scopeItems.accountOnly'), value: SelectorScope.CURRENT },
        { label: getString('rbac.scopeItems.accountAll'), value: SelectorScope.INCLUDE_CHILD_SCOPES }
      ]
  }
}

export const getSelectionType = (resourceGroup?: ResourceGroupV2): SelectionType => {
  if (resourceGroup?.resourceFilter?.includeAllResources) {
    return SelectionType.ALL
  }
  return SelectionType.SPECIFIED
}
export const getScopeType = (resourceGroup?: ResourceGroupV2): SelectorScope => {
  if (resourceGroup?.includedScopes?.length) {
    for (const scope of resourceGroup?.includedScopes) {
      if (getScopeFromDTO(resourceGroup) === getScopeFromDTO(scope)) {
        return scope.filter === 'INCLUDING_CHILD_SCOPES' ? SelectorScope.INCLUDE_CHILD_SCOPES : SelectorScope.CURRENT
      } else {
        return SelectorScope.CUSTOM
      }
    }
  }

  return SelectorScope.CURRENT
}

export const getFormattedDataForApi = (
  data: ResourceGroupV2,
  selectionType: SelectionType,
  includedScopes: ScopeSelector[],
  resourceSelectors: ResourceSelectorV2[]
): ResourceGroupV2Request => {
  return {
    resourceGroup: {
      ...data,
      resourceFilter:
        selectionType === SelectionType.ALL
          ? {
              includeAllResources: true
            }
          : {
              resources: resourceSelectors,
              includeAllResources: false
            },
      includedScopes: includedScopes === [] ? getIncludedScopes(SelectorScope.CURRENT, data) : includedScopes
    }
  }
}

export const getFilteredResourceTypes = (
  data: ResponseResourceTypeDTO | null,
  selectedScope: SelectorScope
): ResourceType[] => {
  const resourceTypes = data?.data?.resourceTypes || []
  return resourceTypes?.reduce((acc: ResourceType[], value) => {
    if (
      value.name &&
      RbacFactory.isRegisteredResourceType(value.name as ResourceType) &&
      ((selectedScope === SelectorScope.CURRENT && value.validatorTypes?.includes(ValidatorTypes.BY_RESOURCE_TYPE)) ||
        selectedScope !== SelectorScope.CURRENT)
    ) {
      acc.push(value.name as ResourceType)
    }
    return acc
  }, [])
}

export const cleanUpResourcesMap = (
  resourceTypes: ResourceType[],
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  selectionType: SelectionType
): Map<ResourceType, string | string[]> => {
  const types = [...resourceTypes]
  selectedResourcesMap.forEach((_value, key) => {
    if (!types.includes(key)) {
      selectedResourcesMap.delete(key)
    } else {
      types.splice(types.indexOf(key), 1)
      selectedResourcesMap.set(key, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    }
  })
  if (selectionType === SelectionType.ALL) {
    types.map(resourceType => {
      selectedResourcesMap.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    })
  }
  return selectedResourcesMap
}

export const getDefaultIncludedScope = (
  accountIdentifier: string,
  orgIdentifier?: string,
  projectIdentifier?: string,
  includedScopes?: ScopeSelector[]
): ScopeSelector[] => {
  if (includedScopes?.length) {
    return includedScopes
  }
  return [
    {
      accountIdentifier,
      orgIdentifier,
      projectIdentifier,
      filter: 'EXCLUDING_CHILD_SCOPES'
    }
  ]
}

export const getIncludedScopes = (type: SelectorScope, resourceGroup: ResourceGroupV2): ScopeSelector[] => {
  const { accountIdentifier, orgIdentifier, projectIdentifier } = resourceGroup
  switch (type) {
    case SelectorScope.CUSTOM:
      //Replace with custom scope once it is added
      return []
    case SelectorScope.INCLUDE_CHILD_SCOPES:
      return [
        {
          accountIdentifier,
          orgIdentifier,
          projectIdentifier,
          filter: 'INCLUDING_CHILD_SCOPES'
        }
      ]
    default:
      return [
        {
          accountIdentifier,
          orgIdentifier,
          projectIdentifier,
          filter: 'EXCLUDING_CHILD_SCOPES'
        }
      ]
  }
}
