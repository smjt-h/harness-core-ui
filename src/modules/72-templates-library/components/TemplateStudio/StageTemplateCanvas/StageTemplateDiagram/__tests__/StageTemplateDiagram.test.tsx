/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { act, fireEvent, render } from '@testing-library/react'
import produce from 'immer'
import { TestWrapper } from '@common/utils/testUtils'
import {
  approvalStageTemplateMock,
  stageTemplateMock
} from '@templates-library/components/TemplateStudio/__tests__/stateMock'
import { PipelineContext } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
import pipelineContextMock from '@pipeline/components/PipelineStudio/RightDrawer/__tests__/stateMock'
import {
  DefaultNewStageId,
  DefaultNewStageName
} from '@templates-library/components/TemplateStudio/StageTemplateCanvas/StageTemplateForm/StageTemplateForm'
import { StageType } from '@pipeline/utils/stageHelpers'
import { StageTemplateDiagram } from '../StageTemplateDiagram'

jest.mock('@common/components/YAMLBuilder/YamlBuilder')
jest.mock('@wings-software/monaco-yaml/lib/esm/languageservice/yamlLanguageService', () => ({
  getLanguageService: jest.fn()
}))

const contextMock = produce(pipelineContextMock, draft => {
  draft.stagesMap = {
    Deployment: {
      name: 'Deploy',
      type: StageType.DEPLOY,
      icon: 'cd-main',
      iconColor: 'var(--pipeline-deploy-stage-color)',
      isApproval: false,
      openExecutionStrategy: true
    },
    Approval: {
      name: 'Approval',
      type: StageType.APPROVAL,
      icon: 'approval-stage-icon',
      iconColor: 'var(--pipeline-approval-stage-color)',
      isApproval: true,
      openExecutionStrategy: false
    }
  }
})

describe('<StageTemplateDiagram /> tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('should render plus button when stage is not selected', async () => {
    const context = produce(contextMock, draft => {
      draft.getStageFromPipeline = _stageId => {
        return {}
      }
    })
    const { container } = render(
      <TestWrapper>
        <PipelineContext.Provider value={context}>
          <StageTemplateDiagram />
        </PipelineContext.Provider>
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
    const node = container.querySelector('[data-nodeid="create-node"] .defaultNode') as HTMLElement
    expect(node).toBeDefined()
    let dynamicPopover = document.querySelector('[class*="dynamicPopover"]')
    expect(dynamicPopover).toBeNull()
    await act(async () => {
      fireEvent.click(node)
    })
    dynamicPopover = document.querySelector('[class*="dynamicPopover"]')
    expect(dynamicPopover).toBeDefined()
  })

  test('should render stage button when stage is selected', async () => {
    const context = produce(contextMock, draft => {
      draft.getStageFromPipeline = jest.fn().mockReturnValue({
        stage: {
          stage: {
            ...stageTemplateMock.spec,
            name: DefaultNewStageName,
            identifier: DefaultNewStageId
          }
        }
      })
    })

    const { container } = render(
      <TestWrapper>
        <PipelineContext.Provider value={context}>
          <StageTemplateDiagram />
        </PipelineContext.Provider>
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test.only('should render stage button when approval  is selected', async () => {
    const context = produce(contextMock, draft => {
      draft.getStageFromPipeline = jest.fn().mockReturnValue({
        stage: {
          stage: {
            ...approvalStageTemplateMock.spec,
            name: DefaultNewStageName,
            identifier: DefaultNewStageId
          }
        }
      })
    })

    const { container } = render(
      <TestWrapper>
        <PipelineContext.Provider value={context}>
          <StageTemplateDiagram />
        </PipelineContext.Provider>
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
