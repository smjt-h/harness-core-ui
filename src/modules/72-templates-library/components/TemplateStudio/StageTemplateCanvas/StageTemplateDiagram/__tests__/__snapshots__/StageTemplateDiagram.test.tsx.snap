// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageTemplateDiagram /> tests should render plus button when stage is not selected 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main container StyledProps--background-form StyledProps--padding-left-xxxlarge StyledProps--padding-right-xxxlarge"
    style="width: 100%;"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-flex-start"
      style="height: 100%;"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600"
      >
        Stage Type
      </p>
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main normalLayout"
        >
          <div
            class="StyledProps--font StyledProps--main"
            data-nodeid="create-node"
          >
            <div
              class="defaultNode createNode create-node"
              style="margin-left: -19px;"
            >
              <div
                class="defaultCard createNew createNewModal"
                style="margin-top: 12px; width: 90px; height: 40px;"
              >
                <div>
                  <span
                    class="bp3-icon bp3-icon-plus"
                    icon="plus"
                  >
                    <svg
                      data-icon="plus"
                      fill="var(--diagram-add-node-color)"
                      height="22"
                      viewBox="0 0 20 20"
                      width="22"
                    >
                      <desc>
                        plus
                      </desc>
                      <path
                        d="M16 9h-5V4c0-.55-.45-1-1-1s-1 .45-1 1v5H4c-.55 0-1 .45-1 1s.45 1 1 1h5v5c0 .55.45 1 1 1s1-.45 1-1v-5h5c.55 0 1-.45 1-1s-.45-1-1-1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  <div>
                    <div
                      style="visibility: visible;"
                    />
                    <div
                      style="visibility: visible;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <p
            class="StyledProps--font StyledProps--main stageType StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600"
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<StageTemplateDiagram /> tests should render stage button when approval  is selected 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main container StyledProps--background-form StyledProps--padding-left-xxxlarge StyledProps--padding-right-xxxlarge"
    style="width: 100%;"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-flex-start"
      style="height: 100%;"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600"
      >
        Stage Type
      </p>
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main approvalLayout"
        >
          <div
            class="StyledProps--font StyledProps--main"
            data-nodeid="create-node"
          >
            <div
              class="defaultNode"
            >
              <div
                draggable="false"
              >
                <div
                  class="defaultCard diamond createNewModal"
                  style="width: 57px; height: 57px; opacity: 1; border-width: 1px; border-style: solid;"
                >
                  <div
                    class="execution-running-animation"
                  />
                  <span
                    class="bp3-icon StyledProps--main"
                    data-icon="approval-stage-icon"
                  >
                    <svg
                      height="28"
                      viewBox="0 0 33 33"
                      width="28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M24.695 13.426l2.025 2.026a1 1 0 010 1.414l-9.871 9.871a1 1 0 01-1.414 0L5.248 16.551a1 1 0 010-1.414l9.844-9.844a1 1 0 011.428.014l3.95 4.106 1.441-1.386-3.95-4.106a3 3 0 00-4.283-.042l-9.844 9.844a3 3 0 000 4.242L14.02 28.152a3 3 0 004.243 0l9.871-9.872a3 3 0 000-4.243l-2.025-2.025-1.415 1.414zm1.4-4.114a1 1 0 00-1.413-1.414l-9.271 9.27-3.783-3.782a1 1 0 10-1.414 1.414l5.197 5.197L26.096 9.312z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
                <p
                  class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-size-normal StyledProps--font-align-center StyledProps--margin-top-xsmall StyledProps--padding-xsmall"
                  style="cursor: pointer; line-height: 1.6; overflow-wrap: normal; word-break: keep-all; --text-line-clamp: 2; width: 125px;"
                >
                  
                </p>
              </div>
            </div>
          </div>
          <p
            class="StyledProps--font StyledProps--main stageType StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600"
          >
            Approval
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<StageTemplateDiagram /> tests should render stage button when stage is selected 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main container StyledProps--background-form StyledProps--padding-left-xxxlarge StyledProps--padding-right-xxxlarge"
    style="width: 100%;"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-flex-start"
      style="height: 100%;"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600"
      >
        Stage Type
      </p>
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main normalLayout"
        >
          <div
            class="StyledProps--font StyledProps--main"
            data-nodeid="create-node"
          >
            <div
              class="defaultNode default-node"
              style="margin-left: -19px;"
            >
              <div
                class="defaultCard createNewModal"
                draggable="false"
                style="width: 90px; height: 40px; margin-top: 12px; cursor: pointer; opacity: 1; border-width: 1px; border-style: solid;"
              >
                <div
                  class="execution-running-animation"
                />
                <div
                  style="visibility: visible;"
                />
                <div
                  style="visibility: visible;"
                />
                <span
                  class="bp3-icon StyledProps--main secondaryIcon"
                  data-icon="command-echo"
                >
                  <svg
                    height="8"
                    viewBox="0 0 14 11"
                    width="8"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.2 2.8c0-.4-.3-.7-.7-.7-.2 0-.4 0-.5.2L.2 5c-.1.1-.2.3-.2.5s.1.4.2.5L3 8.7c.1.1.3.2.5.2.4 0 .7-.3.7-.7 0-.2-.1-.4-.2-.5L1.7 5.5 4 3.2c.1-.1.2-.3.2-.4zM8.4 0c-.3 0-.6.2-.7.5l-2.8 9.6v.2c0 .4.3.7.7.7.3 0 .6-.2.7-.5L9.1.9V.7c0-.4-.3-.7-.7-.7zm5.4 5L11 2.3c-.1-.1-.3-.2-.5-.2-.4 0-.7.3-.7.7 0 .2.1.4.2.5l2.3 2.3L10 7.8c-.1.1-.2.3-.2.4 0 .4.3.7.7.7.2 0 .4-.1.5-.2L13.8 6c.1-.1.2-.3.2-.5s-.1-.4-.2-.5z"
                      fill="#4d5969"
                    />
                  </svg>
                </span>
              </div>
              
            </div>
          </div>
          <p
            class="StyledProps--font StyledProps--main stageType StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600"
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
